sdk/python/feast/feature_store.py

```diff
@@ -176,3 +240,113 @@ def _get_requested_feature_views(
+    if feature_view.input.field_mapping is not None:
+        reverse_field_mapping = {
+            v: k for k, v in feature_view.input.field_mapping.items()
+        }
+        event_timestamp_column = (
+            reverse_field_mapping[event_timestamp_column]
+            if event_timestamp_column in reverse_field_mapping.keys()
+            else event_timestamp_column
+        )
+        created_timestamp_column = (
+            reverse_field_mapping[created_timestamp_column]
+            if created_timestamp_column is not None
+            and created_timestamp_column in reverse_field_mapping.keys()
+            else created_timestamp_column
+        )
+        entity_names = [
+            reverse_field_mapping[col] if col in reverse_field_mapping.keys() else col
+            for col in entity_names
+        ]
+        feature_names = [
+            reverse_field_mapping[col] if col in reverse_field_mapping.keys() else col
+            for col in feature_names
+        ]
+    return (
+        entity_names,
+        feature_names,
+        event_timestamp_column,
+        created_timestamp_column,
+    )
+
+
+def _run_forward_field_mapping(
+    table: pyarrow.Table, field_mapping: Dict[str, str],
+) -> pyarrow.Table:
+    # run field mapping in the forward direction
+    cols = table.column_names
+    mapped_cols = [
+        field_mapping[col] if col in field_mapping.keys() else col for col in cols
+    ]
+    table = table.rename_columns(mapped_cols)
+    return table
+
+
+def _convert_arrow_to_proto(
+    table: pyarrow.Table, feature_view: FeatureView
+) -> List[Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]]:
+    rows_to_write = []
+    for row in zip(*table.to_pydict().values()):
+        entity_key = EntityKeyProto()
+        for entity_name in feature_view.entities:
+            entity_key.entity_names.append(entity_name)
+            idx = table.column_names.index(entity_name)
+            value = _convert_to_proto(row[idx])
+            entity_key.entity_values.append(value)
+        feature_dict = {}
+        for feature in feature_view.features:
+            idx = table.column_names.index(feature.name)
+            value = _convert_to_proto(row[idx])
+            feature_dict[feature.name] = value
+        event_timestamp_idx = table.column_names.index(
+            feature_view.input.event_timestamp_column
+        )
+        event_timestamp = row[event_timestamp_idx]
+        if feature_view.input.created_timestamp_column is not None:
+            created_timestamp_idx = table.column_names.index(
+                feature_view.input.created_timestamp_column
+            )
+            created_timestamp = row[created_timestamp_idx]
+        else:
+            created_timestamp = None
+
+        rows_to_write.append(
+            (entity_key, feature_dict, event_timestamp, created_timestamp)
+        )
+    return rows_to_write
+
+
+def _convert_to_proto(value: Any) -> ValueProto:
```

Comment:
I believe we're still missing types here. Also, we should move this method to `value_types.py` and give it a descriptive name. I would also confirm whether we dont already have a type map for this mapping.